Notifications

kristofbc opened the pull request:
| Question               | Answer
| ---------------------- | -
| Bugs correction ?      | no
| New features ?         | yes
| Related issues         | close #26

---

##### What have you changed  ?
The project page with the new notification button

##### How did you do that ?
A new NotificationService was added to communicate with the new /notification endpoint of the server;
A new page was also added and is used to display read and unread notification that you are the recipient.

**Important, it cannot work without the back-end: https://github.com/ColabTask/api-tasklist/pull/17**

kristofbc committed:
Adding the notification model, service and interface

kristofbc committed:
removing the user id from the notification endpoint

RignonNoel commented on tasklist/src/pages/listNotification/listNotification.ts:
```diff
@@ -0,0 +1,69 @@
+    // Mark notification as read or unread
+    let handler;
+    if( notification.is_read ) {
+      handler = this.notificationService.markNotificationAsUnread(notification.id);
```
I don't understand (but i've no tried to run the code for the moment so it's maybe my bad).

This line say (and the line 51 too) we can put a notification already `read` as `unread` ? Beacause it's seems strange to me to allow this action.

RignonNoel commented on tasklist/src/services/notificationService.ts:
```diff
@@ -0,0 +1,71 @@
+  constructor(public storage: Storage, protected http:Http, private _config:Config) {
+    this.serverURL = _config.get('apiUrl');
+    storage.get('user_id').then(
+      (user_id) => {
```
I think it's an artefact of your first urls architecture on the API because you don't need the user_id in this line.

kristofbc replied:
Indeed, my bad!

RignonNoel reviewed (changes requested):
After full review it seems you want to include the possibility to put a notification as Unread. I really not against this choice but i don't understant the value, for me it's seems just strange.

Otherwise all the rest seems good to me, just check this little artefact to be sure your good with that.

Good work 👍

kristofbc committed:
Removing unused configuration

kristofbc commented:
@RignonNoel I removed the unused check!
Also, I added the read/unread feature thinking that, maybe you've 'read' a notification but then, you want to save it for later, so you 'unread' it. Like a email!

RignonNoel reviewed (approved):
Yeah, it's not a bad idea. ;)

kristofbc committed:
Merge branch 'develop' into ft-notification-integration
